### DISCUSSION TITLE
How to explicitly return `Unknown`?

### DISCUSSION THREAD

**Discussion by Azureblade3808** [microsoft/pyright#4867] [Q&A] [answered]
I'd like to have an overloaded function/method to return `Unknown` instead of `Any` under some condition, to make the result type-check when being used later. I'm currently not sure how to properly express the return type as `Unknown`.

Sample --
```python
# pyright: reportUnknownArgumentType=true
# pyright: reportUnknownMemberType=false

from __future__ import annotations

from typing import Any, Generic, TypeVar, overload
from typing_extensions import Never

T = TypeVar("T")

class C(Generic[T]):
    @overload
    def f(self: C[Never]) -> Unknown: ...  # pyright: ignore[reportUndefinedVariable, reportUnknownVariableType]
    @overload
    def f(self: C[bool]) -> int: ...
    @overload
    def f(self: C[int]) -> int: ...
    @overload
    def f(self) -> T: ...
    def f(self) -> Any: ...

def accept_int(x: int) -> None: ...

accept_int(C().f())  # Expected to emit an error.
```
If the first overload of `C.f` were changed to return `Any`, or were totally removed, the last line would not get reported.

**Comment by erictraut** [comment, marked as answer]
`Unknown` is not part of the Python type system, so there's no explicit way to describe it. By definition, it's an "implicit `Any`". You can produce an `Unknown` by omitting a type argument or an annotation.

**Reply by Azureblade3808** [reply]
I've found that for an overload, while omitting return type translates to `Unknown` in a `.pyi` file, it translates to `None` in a `.py` file. So I think I will need to use a custom `Unknown` then. Thank you for your help.

**Reply by erictraut** [reply]
Ah, that's a bug. Pyright shouldn't attempt to infer the return type of an function or method that's decorated as `@overload`. I've introduced a fix for this. It will be included in the next release.
